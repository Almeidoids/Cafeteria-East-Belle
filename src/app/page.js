"use client"

import Image from "next/image";
import { useRef, useEffect, useState } from "react";
import { useRouter, usePathname } from "next/navigation";
import Link from "next/link";

//estilos
import "bootstrap/dist/css/bootstrap.min.css"
import styles from "../../style/page.module.css"

//fontes
import { candal } from "../../public/fonts/fonts";

//Componentes
import Menu from "../../components/menu";
import CardIndex from "../../components/cardIndex";
import Rodape from "../../components/rodape";
import Receitas from "../../components/receitas";

//Constantes
import receitas from "../../constants/receitas";
import produtos from "../../constants/produtos";
import CardItens from "../../components/cardItens";
import Carrossel from "../../components/carrossel";

export default function Index() {

  const refMenu = useRef(0);
  const refImage = useRef(0);
  const location = usePathname();
  const [index, setIndex] = useState(0);
  const selledList = produtos.toSorted((a, b) => b.buyed - a.buyed);
  const [moreSelled, setMoreSelled] = useState(null);

  useEffect(() => {
    const observer = new IntersectionObserver(
      ([entry]) => {
        if (entry.isIntersecting) {
          refMenu.current.style.position = "static";
          refMenu.current.style.top = 0;          
        }
      },
    {root: null, threshold: 0.0}
    )

    function roll() {
      const menuRect = refMenu.current.getBoundingClientRect();
      const isAtTop = menuRect.top <= 0;
      
      if (isAtTop) {
        refMenu.current.style.position = "fixed";
        refMenu.current.style.top = 0;
        refMenu.current.style.zIndex = 10;
      }
      else {
        observer.observe(refImage.current);
      };
    }
    
    window.addEventListener("scroll", roll);

    return () => {
      window.removeEventListener("scroll", roll)

      observer.disconnect
    }
  }, []);

  useEffect(() => {
    setMoreSelled(selledList.map(item => {
      if(selledList.indexOf(item) < 12) {
      return (
          <CardItens
            key = {item.key}
            offer = {item.offer}
            image = {item.image}
            title = {item.name}
            original = {item.originalp}
            link = {`/produtos/${item.key}`}
          />
      )};
    }))
  }, []);

  const revenue = receitas.map(item => {
    return (
      <CardIndex
          key = {receitas.indexOf(item)}
          name = {item.name}
          image = {item.image}
          description = {item.description}
          onClick = {() => changeLink(`/item${receitas.indexOf(item)}`, setIndex)}
      />
    );
  })

  const offers = produtos.map(item => {
    if(item.key <= 12 && item.offer !== 0) {
      return (
          <CardItens
            key = {item.key}
            offer = {item.offer}
            image = {item.image}
            title = {item.name}
            original = {item.originalp}
            link = {`/produtos/${item.key}`}
          />
      )
    };
  })

  return (
    <div style = {candal.style}>
      <div className = "fundoBanner">
        <Image
          src = "/banner/banner.jpg"
          alt = "Banner CafÃ©"
          priority
          className = {styles.banner}
          ref = {refImage}
          width = {1000}
          height = {1000}
        />
      </div>
      <Menu ref = {refMenu}/>
      <h1 className = {styles.titulos}>Receitas</h1>
      <div className = {styles.receitas}>
        {revenue}
      </div>

      {location !== `/` &&
        <Receitas style = {{display: "block"}} close = {() => changeLink("/")} i = {index} />
      } 

      <h4 style = {{color: "#2b061e !important"}}>Ofertas: </h4>
      <Carrossel>
          {offers}
      </Carrossel>
      <h4 style = {{color: "#2b061e !important"}}>Mais Vendidos: </h4>
      <Carrossel>
        {moreSelled}
      </Carrossel>
      <Rodape />
    </div>  
  );
}

export function changeLink(url, setIndex) {
  const arrayurl = url.slice("");
  url !== "/" ?
  setIndex(Number(arrayurl[arrayurl.length -1 ])) : arrayurl;

  if (typeof window === "undefined") {
    return;
  }
  else {
    window.history.pushState(null, "", url);
  }
}