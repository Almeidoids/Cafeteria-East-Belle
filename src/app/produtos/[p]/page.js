import Image from "next/image";
import Form from "next/form";

//Componentes
import Menu from "../../../../components/menu"
import Rodape from "../../../../components/rodape"
import Formulario from "../../../../components/formulario";
import ModalConfirm from "../../../../components/modalConfirm";

//Fontes
import { candal, caveat } from "../../../../public/fonts/fonts"

//Estilos
import styles from "../../../../style/p.module.css"
import "bootstrap/dist/css/bootstrap.min.css";
import "bootstrap-icons/font/bootstrap-icons.css";

//Constantes
import produtos from "../../../../constants/produtos";

export default async function p({params}) {
    const {p} = await params
    const product = produtos[Number(p - 1)];
    const subtract = ((product.originalp * product.offer) / 100).toFixed(2);

    return (
        <div style = {candal.style} className = {caveat.variable}>
            <Menu style = {{position: "fixed", top: 0}} />
            <div className = {styles.alinhamento}>
                <div className = {styles.alinhamentoProduto}>
                    <h1 className = {styles.titulo}>{product.name}</h1>
                    {product.offer !== 0 &&
                        <h5 style = {{fontSize: 14, color: "#604456"}} className = {styles.titulo}>Com {product.offer}% de desconto</h5>
                    }
                    <div style = {{marginTop: "5%"}}>
                        <p className = {styles.texto}>{product.text}</p>
                    </div>
                    <div className = {styles.txtcompra}>
                        {product.offer !==0 &&
                            <p className = {styles.oferta}>{product.originalp}</p>
                        }
                        <p className = {styles.preco}>{(product.originalp - subtract).toFixed(2)}</p>
                        <Formulario product = {product} />
                    </div>
                </div>
                <div>
                    <Image
                        src = {product.imagep}
                        alt = {product.name}
                        height = {400}
                        width = {400}
                        className = {styles.imagem}
                    />
                </div>
            </div>
            <Rodape className = {styles.footerR} />
        </div>
    )
}