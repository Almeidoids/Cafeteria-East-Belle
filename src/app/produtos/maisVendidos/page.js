"use client"

import { candal } from "../../../../public/fonts/fonts";

//CÃ³digos
import produtos from "../../../../constants/produtos";

//Componentes
import Menu from "../../../../components/menu";
import Rodape from "../../../../components/rodape";
import CardItens from "../../../../components/cardItens";
//Estilos
import "bootstrap/dist/css/bootstrap.min.css";
import styles from "../../../../style/ofertas.module.css"
import { useEffect, useState } from "react";


export default function maisVendidos() {
    const [lista,  setLista] = useState(produtos.toSorted((a, b) => b.buyed - a.buyed));
    const [maisVendidos, setMaisVendidos] = useState();

    useEffect(() => {
        setLista(produtos.toSorted((a, b) => b.buyed - a.buyed));
        setMaisVendidos(lista.map(item => {
            return (
                <CardItens
                    key = {item.key}
                    offer = {item.offer}
                    image = {item.image}
                    title = {item.name}
                    original = {item.originalp}
                    link = {`/produtos/${item.key}`}
                />
            );
        }))
    }, [produtos])

    return(
        <div style = {candal.style}>
            <Menu style = {{position: "fixed", zIndex: 2}} />
            <div style = {{justifyItems: "center"}}>
                <div className = {styles.organizacaoProdutos}>
                    {maisVendidos}
                </div>
            </div>
            <Rodape style = {{marginTop: "5%"}}/>
        </div>
    )
}