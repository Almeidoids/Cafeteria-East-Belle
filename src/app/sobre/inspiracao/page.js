"use client"

import { useRef, useEffect } from "react"
import Image from "next/image";

//Componentes
import Menu from "../../../../components/menu"
import Rodape from "../../../../components/rodape";

//Estilos
import styles from "../../../../style/sobre.module.css";

//Fontes
import { candal } from "../../../../public/fonts/fonts";

export default function Inspiracao() {
    const menu = useRef(null);

    useEffect(() => {
        menu.current.style.position = "fixed";
        menu.current.style.top = 0;
        menu.current.style.zIndex = 2;
    }, []);

    return (
        <div style = {candal.style}>
            <Menu ref = {menu} />
            <div className = {styles.organizacao}>
                <div className = {styles.fundotxtEsq}>
                    <h4 className = "title2">A Magia por Trás da East-Belle: Uma Viagem Cinematográfica</h4>
                    <p className = "text">Se você já se sentiu como se estivesse em um filme, daqueles que te fazem sorrir e suspirar ao mesmo tempo, então vai entender a nossa inspiração. A Cafeteria East-Belle nasceu de um desejo profundo de criar um espaço onde a magia do cinema se encontrasse com o aconchego de um café.</p>
                    <h4 className = "title2">"O Fabuloso Destino de Amelie Poulain": A Faísca da Inspiração</h4>
                    <p className = "text">Assim como no filme "O Fabuloso Destino de Amelie Poulain", buscamos referências em cada detalhe:</p>
                    <ul className = "list">
                        <li>As cores vibrantes e alegres: que nos remetem à energia contagiante da personagem principal, Amelie.</li>
                        <li>Os objetos vintage e excêntricos: que nos transportam para um universo paralelo de sonhos e fantasias, assim como no quarto de Amelie.</li>
                        <li>A trilha sonora francesa: que embala cada gole de café e cada conversa, criando uma atmosfera única e convidativa.</li>
                    </ul>
                </div>
                <Image 
                    src = "/images/sobre/inspiracao/filme.jpg"
                    alt = "cena do Filme: O Fabuloso Destino de Amelie Poulain."
                    width = {1000}
                    height = {1000}
                    className = {styles.imagem}
                />
                <Image 
                    src = "/images/sobre/cafeteriaEastBelle.jpg"
                    alt = "Uma das unidades da Cafeteria East-Belle"
                    width = {1000}
                    height = {1000}
                    className = {styles.imagem}
                />
                <div className = {styles.fundotxtEsq} style = {{marginBottom: "5%"}}>
                    <h4 className = "title2">Mais do que um Café, um Portal para Outro Mundo</h4>
                    <p className = "text">Mais do que um simples café, a East-Belle é um portal para outro mundo, um refúgio onde a realidade se mistura com a fantasia, onde os clientes se sentem como personagens de um filme. Assim como Amelie transformava a vida das pessoas ao seu redor, queremos que a East-Belle transforme o seu dia, trazendo mais leveza, alegria e sabor.</p>
                    <h4 className = "title2">Um Convite à Magia</h4>
                    <p className = "text">Seja para um café da manhã especial, um encontro com amigos ou um momento de solitude, a East-Belle está sempre de portas abertas para te receber e te transportar para um universo mágico, onde os sonhos se tornam realidade.</p>
                </div>
            </div>
            <div className = {styles.fundotxtEsq} style = {{backgroundColor: "#F9DBBD", justifySelf: "center", width: "auto"}}>
                <h1 className = "title">Venha viver essa experiência e se encantar com a magia da East-Belle!</h1>
            </div>
            <Rodape />
        </div>
    )
}