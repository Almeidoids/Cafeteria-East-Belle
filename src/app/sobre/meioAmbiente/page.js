"use client"

import { useRef, useEffect } from "react";
import { candal, caveat } from "../../../../public/fonts/fonts";
import Image from "next/image";

//Componentes
import Menu from "../../../../components/menu"
import Rodape from "../../../../components/rodape"

//Estilos
import styles from "../../../../style/sobre.module.css";

export default function meioAmbiente() {
    const menu = useRef(null);
    
        useEffect(() => {
            menu.current.style.position = "fixed";
            menu.current.style.top = 0;
            menu.current.style.zIndex = 2;
        }, []);

    return (
        <div style = {candal.style}>
            <Menu ref = {menu} />
            <div className = {`${styles.organizacao}`}>
                <div className = {`${styles.fundotxtEsq} ${styles.sumidao} section-1`}>
                    <p className = {`${styles.paragrafo} text`} style = {caveat.style}>Na Cafeteria East-Belle, acreditamos que a sustentabilidade é fundamental para o futuro do nosso planeta. É por isso que estamos comprometidos em proteger o meio ambiente em todas as nossas operações.</p>
                    <h4 className = {`${styles.titulo} title2`}>Nossas Práticas Sustentáveis</h4>
                    <ul className = "list">
                        <li>Uso de Energia Renovável: Utilizamos energia solar para alimentar nossa cafeteria, reduzindo nossa pegada de carbono e economizando recursos naturais.</li>
                        <li>Reciclagem e Compostagem: Separamos e reciclamos todos os nossos resíduos, incluindo papel, plástico, vidro e alumínio. Também compostamos nossos resíduos orgânicos, como borra de café e restos de comida.</li>
                        <li>Uso de Produtos Sustentáveis: Servimos apenas produtos sustentáveis, como café orgânico, leite de origem animal e alimentos frescos e locais.</li>
                        <li>Educação Ambiental: Estamos comprometidos em educar nossos clientes sobre a importância da sustentabilidade. Oferecemos workshops e eventos sobre temas ambientais e incentivamos nossos clientes a adotarem práticas sustentáveis em suas vidas.</li>
                    </ul>
                    <h4 className = {`${styles.titulo} title2`}>Nosso Impacto</h4>
                    <p className = {`${styles.paragrafo} text`}>Estamos orgulhosos de ter um impacto positivo no meio ambiente. Desde que abrimos nossa cafeteria, reduzimos nossa pegada de carbono em mais de 50% e reciclamos mais de 90% de nossos resíduos.</p>
                </div>
                <div className = {`${styles.fundotxtPeq2} section-1`}>
                    <p className = {`${styles.paragrafo} text`} style = {caveat.style}>Na Cafeteria East-Belle, acreditamos que a sustentabilidade é fundamental para o futuro do nosso planeta. É por isso que estamos comprometidos em proteger o meio ambiente em todas as nossas operações.</p>
                </div>
                <div className = {`${styles.fundotxtPeq2}`}>
                    <h4 className = {`${styles.titulo} title2`}>Nossas Práticas Sustentáveis</h4>
                    <ul className = "list">
                        <li>Uso de Energia Renovável: Utilizamos energia solar para alimentar nossa cafeteria, reduzindo nossa pegada de carbono e economizando recursos naturais.</li>
                        <li>Reciclagem e Compostagem: Separamos e reciclamos todos os nossos resíduos, incluindo papel, plástico, vidro e alumínio. Também compostamos nossos resíduos orgânicos, como borra de café e restos de comida.</li>
                        <li>Uso de Produtos Sustentáveis: Servimos apenas produtos sustentáveis, como café orgânico, leite de origem animal e alimentos frescos e locais.</li>
                        <li>Educação Ambiental: Estamos comprometidos em educar nossos clientes sobre a importância da sustentabilidade. Oferecemos workshops e eventos sobre temas ambientais e incentivamos nossos clientes a adotarem práticas sustentáveis em suas vidas.</li>
                    </ul>
                </div>
                <div className = {`${styles.fundotxtPeq2}`}>
                    <h4 className = {`${styles.titulo} title2`}>Nosso Impacto</h4>
                    <p className = {`${styles.paragrafo} text`}>Estamos orgulhosos de ter um impacto positivo no meio ambiente. Desde que abrimos nossa cafeteria, reduzimos nossa pegada de carbono em mais de 50% e reciclamos mais de 90% de nossos resíduos.</p>
                </div>
                <Image 
                    src = "/images/sobre/meioAmbiente/cafeteria.jpg"
                    alt = "Uma das unidades da Cafeteria East-Belle"
                    width = {1000}
                    height = {1000}
                    className = {styles.imagem}
                />
                <Image 
                    src = "/images/sobre/meioAmbiente/floresta.jpg"
                    alt = "Imagem de uma arvore"
                    width = {1000}
                    height = {1000}
                    className = {styles.imagem}
                />
                <div className = {`${styles.fundotxtEsq} section-2`}>
                    <h4 className = {`${styles.titulo} title2`}>Como Você Pode Ajudar</h4>
                    <p className = {`${styles.paragrafo} text`}>Existem muitas maneiras de ajudar a proteger o meio ambiente. Aqui estão algumas dicas:</p>
                    <ul className = "list">
                        <li>Recicle e compostagem: Separe e recicle seus resíduos. Você também pode compostar seus resíduos orgânicos.</li>
                        <li>Use menos energia: Desligue as luzes e aparelhos quando não estiver usando. Você também pode usar lâmpadas de LED, que são mais eficientes em termos de energia.</li>
                        <li>Conserve água: Tome banhos mais curtos, conserte vazamentos e use menos água para regar seu jardim.</li>
                        <li>Compre produtos sustentáveis: Escolha produtos que sejam fabricados de forma sustentável e que tenham embalagens recicláveis.</li>
                        <li>Apoie empresas sustentáveis: Apoie empresas que estão comprometidas com a sustentabilidade.</li>
                    </ul>
                       <p className = {`${styles.paragrafo} text`}>Juntos, podemos fazer a diferença para o nosso planeta.</p>
                </div>
                
            </div>
            <Rodape style = {{ScrollSnapAlign: "center"}} />
        </div>
    )
}