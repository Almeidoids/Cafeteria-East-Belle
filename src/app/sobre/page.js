"use client"

import { useEffect, useRef } from "react";
import Image from "next/image";

//Componentes
import Menu from "../../../components/menu"
import Rodape from "../../../components/rodape"

//Estilos
import styles from "../../../style/sobre.module.css";

//Fontes
import { candal } from "../../../public/fonts/fonts";

export default function Sobre() {
    const menu = useRef(null);
    
    useEffect(() => {
        menu.current.style.position = "fixed";
        menu.current.style.top = 0;
        menu.current.style.zIndex = 2;
    }, []);

    return(
        <div style = {candal.style}>
            <Menu ref = {menu} />
            <div className = {styles.organizacao}>
                <div className = {`${styles.fundotxtEsq} ${styles.sumidao}`} style = {{width: "70%"}}>
                    <h4 className = "title2">Cafeteria East-Belle</h4>
                    <p className = "text">Bem-vindo à Cafeteria East-Belle, um lugar onde o charme vintage encontra o conforto moderno. Localizada no coração da cidade, nossa cafeteria é um refúgio acolhedor para aqueles que buscam uma pausa na agitação do dia a dia.</p>
                    <h4 className = "title2">Ambiente Único</h4>
                    <p className = "text">Um lugar onde o charme do passado encontra o conforto do presente. Nosso espaço é decorado com paredes de tijolos aparentes e móveis de madeira, criando um ambiente nostálgico e aconchegante. A iluminação suave e a música tranquila completam o clima perfeito para relaxar e desfrutar de bons momentos.</p>
                </div>
                <div className = {styles.fundotxtPeq}>
                    <h4 className = "title2">Cafeteria East-Belle</h4>
                    <p className = "text">Bem-vindo à Cafeteria East-Belle, um lugar onde o charme vintage encontra o conforto moderno. Localizada no coração da cidade, nossa cafeteria é um refúgio acolhedor para aqueles que buscam uma pausa na agitação do dia a dia.</p>
                    <h4 className = "title2">Ambiente Único</h4>
                    <p className = "text">Um lugar onde o charme do passado encontra o conforto do presente. Nosso espaço é decorado com paredes de tijolos aparentes e móveis de madeira, criando um ambiente nostálgico e aconchegante. A iluminação suave e a música tranquila completam o clima perfeito para relaxar e desfrutar de bons momentos.</p>
                </div>
                <Image 
                    src = "/images/sobre/cafeteriaEastBelle.jpg"
                    alt = "Uma das unidades da Cafeteria East-Belle"
                    width = {1000}
                    height = {1000}
                    className = {styles.imagem}
                    style = {{display: "block"}}
                />
                <div className = {styles.fundotxtEsq}>
                    <h4 className = "title2">Nossa Equipe</h4>
                    <p className = "text">Recepção calorosa e atendimento personalizado. Nossa equipe é dedicada a proporcionar uma experiência memorável, desde a escolha do grão perfeito até o sorriso ao servir o seu café.</p>
                    <h4 className = "title2">Nossas Bebidas</h4>
                    <p className = "text">Café é a nossa paixão! Oferecemos uma seleção de grãos artesanais, torrados com maestria para realçar os sabores mais ricos. Seja um expresso clássico ou um latte art sofisticado, cada xícara é preparada com carinho e atenção. Além do café, temos chás gourmet, sucos naturais e smoothies refrescantes.</p>
                    <h4 className = "title2">Eventos e Comunidade</h4>
                    <p className = "text">Mais do que uma cafeteria, somos um espaço para a comunidade. Organizamos noites de poesia, apresentações musicais ao vivo e workshops de arte, criando um ambiente inspirador onde todos são bem-vindos.</p>
                    <h4 className = "title2">Nossa Missão</h4>
                    <p className = "text">Proporcionar um espaço perfeito para reunir amigos, trabalhar ou simplesmente relaxar com uma boa xícara de café. Venha nos visitar e descubra por que a Cafeteria East-Belle é tão especial.</p>
                </div>
            </div>
            <Rodape />
        </div>
    )
}

