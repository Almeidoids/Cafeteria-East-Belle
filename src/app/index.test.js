import { render, fireEvent, screen } from "@testing-library/react";
import { changeLink } from "./page";
import CardIndex from "../../components/cardIndex";

describe("Mudança de link", () => {
    const setIndex = jest.fn();
    test("Botão para trocar link", () => {
        const component = render(
            <button onClick = {() => changeLink("/1", setIndex)} />
        )
        const button = screen.getByRole("button");
        fireEvent.click(button);
        expect(setIndex).toHaveBeenCalledTimes(1);
        expect(setIndex).toHaveBeenCalledWith(1);
        expect(window.location.href).toMatch("/1");
    });

    test("CardIndex", () => {
        const component = render(
            <CardIndex 
                key = {0}
                name = {"Nome"}
                image = {"/images/affogato.png"}
                description = "AAA"
            />
        )
        expect(component).toMatchSnapshot();
    })
})