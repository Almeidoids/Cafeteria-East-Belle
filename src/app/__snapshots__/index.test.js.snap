// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mudan√ßa de link CardIndex 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
        style="width: 18rem; height: 30rem; background-color: rgb(43, 6, 30); margin: 50px; border-radius: 20px; align-items: center;"
      >
        <div>
          <img
            alt="Nome"
            class="cafes"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Faffogato.png&w=640&q=75"
            srcset="/_next/image?url=%2Fimages%2Faffogato.png&w=256&q=75 1x, /_next/image?url=%2Fimages%2Faffogato.png&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
        </div>
        <div
          style="height: 40%;"
        >
          <h2
            class="titulos"
          >
            Nome
          </h2>
          <p
            class="textos"
            style="font-family: fontFamily;"
          >
            AAA
          </p>
        </div>
        <button
          class="btn btn-primary"
          style="background-color: rgb(64, 201, 162); margin-bottom: 10px;"
          type="button"
        >
          Veja Mais
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
      style="width: 18rem; height: 30rem; background-color: rgb(43, 6, 30); margin: 50px; border-radius: 20px; align-items: center;"
    >
      <div>
        <img
          alt="Nome"
          class="cafes"
          data-nimg="1"
          decoding="async"
          height="200"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Faffogato.png&w=640&q=75"
          srcset="/_next/image?url=%2Fimages%2Faffogato.png&w=256&q=75 1x, /_next/image?url=%2Fimages%2Faffogato.png&w=640&q=75 2x"
          style="color: transparent;"
          width="200"
        />
      </div>
      <div
        style="height: 40%;"
      >
        <h2
          class="titulos"
        >
          Nome
        </h2>
        <p
          class="textos"
          style="font-family: fontFamily;"
        >
          AAA
        </p>
      </div>
      <button
        class="btn btn-primary"
        style="background-color: rgb(64, 201, 162); margin-bottom: 10px;"
        type="button"
      >
        Veja Mais
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
